### PR TITLE
Add <code> if it's missing

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes [#16445](https://github.com/vector-im/element-web/issues/16445)

### CODE REVIEW

**Review by jryans — CHANGES_REQUESTED** [review]
Hmm, I'm not sure how we actually want to approach this... Why do we depend on having a `code` element at all? In a normal, simple HTML page, a `pre` without a `code` is good enough.

Is there an alternate approach where we stop requiring `code` to be present?

**Comment by SimonBrandner** [comment]
@jryans, I think I should be able to work around this, but I am not sure yet. I'll give it a go

File: src/components/views/messages/TextualBody.js [hunk 1]
```diff
@@ -128,6 +132,13 @@ export default class TextualBody extends React.Component {
         }
     }
 
+    _addCodeElement(pre) {
+        const code = document.createElement("code");
+        code.innerHTML = pre.innerHTML;
```
Comment (jryans) [inline src/components/views/messages/TextualBody.js:137]: Let's avoid using `innerHTML` if we can, see https://stackoverflow.com/a/6938316/486978 on how you can wrap an element.

File: src/components/views/messages/TextualBody.js [hunk 2]
```diff
@@ -99,6 +99,10 @@ export default class TextualBody extends React.Component {
                     // If there already is a div wrapping the codeblock we want to skip this.
                     // This happens after the codeblock was edited.
                     if (pres[i].parentNode.className == "mx_EventTile_pre_container") continue;
+                    // Add code element if it's missing
```
Comment (jryans) [inline src/components/views/messages/TextualBody.js:102]:
```suggestion
                    // Add code element if it's missing since we depend on it
```

**Review by jryans — CHANGES_REQUESTED** [review]
Perhaps this is the least worst thing after all. 😄

File: src/components/views/messages/TextualBody.js [hunk 3]
```diff
@@ -128,6 +132,13 @@ export default class TextualBody extends React.Component {
         }
     }
 
+    _addCodeElement(pre) {
+        const code = document.createElement("code");
+        code.append(...pre.childNodes);
+        pre.innerHTML = "";
```
Comment (jryans) [inline src/components/views/messages/TextualBody.js:138]: Is this actually required...? Appending existing nodes to a different place typically _moves_ them with most DOM APIs, so I would expect `pre` to already be empty...

**Review by jryans — CHANGES_REQUESTED** [review]

File: src/components/views/messages/TextualBody.js (hunk 3 above)
Comment (SimonBrandner) [reply src/components/views/messages/TextualBody.js]: Yeah, you're right. Sorry for all the trouble with this...

Comment (jryans) [reply src/components/views/messages/TextualBody.js]: No worries, no need to stress about it. 😄

**Review by jryans — APPROVED** [review]
Thanks for working this, looks good to me! 😄

**Pull request merged**